ui/src/contexts/Application.jsx

```diff
@@ -235,37 +236,241 @@ const setupAMM = async (dispatch, brandToInfo, zoe, board, instanceID) => {
+  );
+};
+
+const setupRUNStake = async (
+  dispatch,
+  instance,
+  board,
+  zoe,
+  RUN_STAKE_NAME,
+) => {
+  const [RUNStakeAPI, RUNStakeTerms, RUNStakeInstallation] = await Promise.all([
+    E(zoe).getPublicFacet(instance),
+    E(zoe).getTerms(instance),
+    E(zoe).getInstallationForInstance(instance),
+  ]);
+  // Get brands.
+  const brands = [
+    RUNStakeTerms.brands.Attestation,
+    RUNStakeTerms.brands.Debt,
+    RUNStakeTerms.brands.Stake,
+  ];
+  const keywords = ['LIEN', 'RUN', 'BLD'];
+  const displayInfos = await Promise.all(
+    brands.map(b => E(b).getDisplayInfo()),
+  );
+
+  const newBrandToInfo = brands.map((brand, i) => {
+    const decimalPlaces = displayInfos[i] && displayInfos[i].decimalPlaces;
+    /** @type { [Brand, BrandInfo]} */
+    const entry = [
+      brand,
+      {
+        assetKind: displayInfos[i].assetKind,
+        decimalPlaces,
+        petname: keywords[i],
+        brand,
+      },
+    ];
+    return entry;
+  });
+  dispatch(mergeBrandToInfo(newBrandToInfo));
+
+  // Suggest instance/installation
+  const [instanceBoardId, installationBoardId] = await Promise.all([
+    E(board).getId(instance),
+    E(board).getId(RUNStakeInstallation),
+  ]);
+  await Promise.all([
+    E(walletP).suggestInstallation(
+      `${RUN_STAKE_NAME}Installation`,
+      installationBoardId,
+    ),
+    E(walletP).suggestInstance(`${RUN_STAKE_NAME}Instance`, instanceBoardId),
+  ]);
+
+  // Watch for loan invitations.
+  watchLoans(dispatch, instanceBoardId);
+
+  // TODO: Get notifier for governedParams.
+  dispatch(
+    setRUNStake({
+      RUNStakeAPI,
+      RUNStakeTerms,
+      instanceBoardId,
+      installationBoardId,
+    }),
+  );
+};
+
 /* eslint-disable complexity, react/prop-types */
 export default function Provider({ children }) {
   const [state, dispatch] = useReducer(reducer, defaultState);
   const { brandToInfo } = state;
 
+  const urlParams = new URLSearchParams(window.location.search);
+  const useGetRUN = urlParams.get('gr') === 'true';
+
   useEffect(() => {
-    const urlParams = new URLSearchParams(window.location.search);
-    const useGetRUN = urlParams.get('gr') === 'true';
     dispatch(setUseGetRUN(useGetRUN));
   }, []);
 
   const retrySetup = async () => {
-    await refreshConfigFromWallet(walletP);
+    await refreshConfigFromWallet(walletP, useGetRUN);
     const {
       INSTALLATION_BOARD_ID,
       INSTANCE_BOARD_ID,
       RUN_ISSUER_BOARD_ID,
       AMM_INSTALLATION_BOARD_ID,
       AMM_INSTANCE_BOARD_ID,
       AMM_NAME,
+      RUNStakeInstance,
+      RUN_STAKE_NAME,
     } = dappConfig;
 
     const zoe = E(walletP).getZoe();
     const board = E(walletP).getBoard();
     try {
-      await Promise.all([
-        setupTreasury(dispatch, brandToInfo, zoe, board, INSTANCE_BOARD_ID),
-        setupAMM(dispatch, brandToInfo, zoe, board, AMM_INSTANCE_BOARD_ID),
-      ]);
+      if (useGetRUN) {
+        await setupRUNStake(
```

Comment:
For now we don't load the treasury or AMM if the RUNStake flag is enabled. This is because the failure mode for when treasury/amm is not deployed on chain is the promise hangs forever, so we want to be able to test RUNStake separately